src/core_modules/capture-core/trackerOffline/trackerOffline.js

```diff
@@ -1,12 +1,21 @@
 // @flow
 import defaultQueue from '@redux-offline/redux-offline/lib/defaults/queue';
-import { getApi } from '../d2/d2Instance';
-import { methods } from './trackerOfflineConfig.const';
+import { effectMethods } from './trackerOffline.const';
+import type { OfflineEffect } from './trackerOffline.types';
+
+export const getEffectReconciler = (() => {
+    const mutateTypeForMethods = {
+        [effectMethods.POST]: 'create',
+        [effectMethods.UPDATE]: 'replace',
+        [effectMethods.DELETE]: 'delete',
+    };
+
+    return (onApiMutate: Function) => ({ url: resource, method, data }: OfflineEffect) => {
+        const type = mutateTypeForMethods[method];
+        return onApiMutate({ resource, type, data });
+    };
+})();
 
-export const effectConfig = (effect: OfflineEffect) => {
-    const { url, method, data } = effect;
-    return getApi()[method](url, data);
```

Comment:
It was not possible to spy on the network requests when running them through the old d2 library. Because of the way headers are set as highlighted [here](https://github.com/cypress-io/cypress/issues/95#issuecomment-658889331). MIght work with Cypress v5 (I can see the issue was closed recently), but v5 is not used by cli-utils-cypress yet. Anyway, I figured this was a good time to shift to the app runtime for our requests that goes through the offline framework.

Some requests are still using the old d2 library, I suggest we replace these continuously (can't get rid of the old d2 library just yet because of dependencies).

Reply:
Definitely better what you did 🦖